crates/next-core/src/params_matcher/mod.rs

```diff
@@ -0,0 +1,117 @@
+use anyhow::{bail, Result};
+use turbo_tasks::{
+    primitives::{OptionStringVc, StringVc},
+    ValueToString, ValueToStringVc,
+};
+use turbopack_node::match_params::{MatchParams, MatchParamsVc, ParamsMatchesVc};
+
+use self::path_regex::{PathRegex, PathRegexBuilder};
+
+mod path_regex;
+
+#[turbo_tasks::value]
+pub(crate) struct NextParamsMatcher {
+    #[turbo_tasks(trace_ignore)]
+    path_regex: PathRegex,
+    prefix: OptionStringVc,
+    suffix: OptionStringVc,
+}
+
+#[turbo_tasks::value_impl]
+impl NextParamsMatcherVc {
+    /// Converts a filename within the server root into a regular expression
+    /// with named capture groups for every dynamic segment.
+    #[turbo_tasks::function]
+    pub async fn new(
+        pathname: StringVc,
+        prefix: OptionStringVc,
+        suffix: OptionStringVc,
```

Comment:
`Option<StringVc>` is also supported as argument to turbo functions, so you could use that